src/core/core.scale.js

```diff
@@ -1129,9 +1129,13 @@ class Scale extends Element {
 
 			if (isHorizontal) {
 				x = pixel;
-				textOffset = position === 'top'
-					? ((!rotation ? 0.5 : 1) - lineCount) * lineHeight
-					: (!rotation ? 0.5 : 0) * lineHeight;
+				if (position === 'top') {
+					textOffset = (Math.sin(rotation) * (lineCount / 2) + 0.5) * lineHeight;
+					textOffset -= (rotation === 0 ? (lineCount - 0.5) : Math.cos(rotation) * (lineCount / 2)) * lineHeight;
```

Comment:
It seems a little suspect to me that we need to handle the `rotation === 0` case separately. Can we remove that and just rely on the general formula? If not, I'm guessing this won't work for small rotation values approaching 0

Reply:
It's because the alignment changes when rotation is 0. That's set a few lines above this

Reply:
Ah, ok. Thanks for looking into this

Reply:
Yeah, it's super old code here. I tried to make top & bottom similar, but the rest of the draw code needs the Y coordinate to be the first line of text so you have to move by more when positioned on top vs the bottom